### PR TITLE
`method_with_annotation` test

### PR DESCRIPTION
**Pull request by pkesseli**
Add e2e test for sources and sinks marked using `method_with_annotation`.

### CODE REVIEW

**Comment by jvillard** [comment]
Hi Pascal, please let me know if you have any issue with signing the CLA.

**Comment by pkesseli** [comment]
> Hi Pascal, please let me know if you have any issue with signing the CLA.

Hi Jules, thank you for your message! My manager, Peter O'Hearn, sent an email to cla@meta.com and asked them to add me to the CLA he submitted before for Lacework to contribute to Meta projects. I messaged cla@meta.com again last week to ask about the status of that, but did not receive a response.

Can you confirm that I have been added to the CLA on behalf of Lacework, and it is now just a matter of me also signing the CLA personally?

**Comment by jvillard** [comment]
Thanks for flagging @pkesseli, I'm following this up internally. We need to add you to the existing CLA, nothing to do on your end.

File: infer/tests/codetoanalyze/java/pulse/issues.exp [hunk 1]
```diff
@@ -163,6 +163,8 @@ codetoanalyze/java/pulse/WriterLeaks.java, codetoanalyze.java.infer.WriterLeaks.
...
 codetoanalyze/java/pulse/WriterLeaks.java, codetoanalyze.java.infer.WriterLeaks.fileWriterNotClosedAfterWriteBad():void, 8, PULSE_RESOURCE_LEAK, no_bucket, ERROR, [allocation part of the trace starts here,allocated by constructor java.io.FileWriter() here,memory becomes unreachable here]
 codetoanalyze/java/pulse/WriterLeaks.java, codetoanalyze.java.infer.WriterLeaks.pipedWriterNotClosedAfterConstructedWithReaderBad():void, 10, PULSE_RESOURCE_LEAK, no_bucket, ERROR, [allocation part of the trace starts here,allocated by constructor java.io.PipedWriter() here,memory becomes unreachable here]
+codetoanalyze/java/pulse/taint/Annotations.java, codetoanalyze.java.pulse.Annotations.markedSourceToMarkedSink():void, 1, TAINT_ERROR, no_bucket, ERROR, [source of the taint here: value returned from String Annotations.markedSource() with kind Simple,passed as argument #1 to void Annotations.markedSink(String) with kind Simple]
```
Comment (geralt-encore) [inline infer/tests/codetoanalyze/java/pulse/issues.exp:166]: For test cases we usually try to follow explicit naming conventions:
- "bad" suffix is for cases where Infer correctly reports an issue
- "good" suffix is for cases where Infer is not supposed to report anything
- "falsePositive" and "falseNegative" for respective misreporting cases

Comment (pkesseli) [reply infer/tests/codetoanalyze/java/pulse/issues.exp]: Added `Bad` and `Ok` suffixes, no false positives in this one.

**Comment by geralt-encore** [comment]
Similar to https://github.com/facebook/infer/pull/1723 I'll be happy to merge it when CLA is ready

**Review by geralt-encore — APPROVED** [review]

**Pull request closed without merging**